modules/highgui/src/window_QT.cpp

```diff
@@ -673,30 +673,24 @@ CV_IMPL void cvSetTrackbarPos(const char* name_bar, const char* window_name, int
 
 CV_IMPL void cvSetTrackbarMax(const char* name_bar, const char* window_name, int maxval)
 {
-    if (maxval >= 0)
+    QPointer<CvTrackbar> t = icvFindTrackBarByName(name_bar, window_name);
+    if (t)
     {
-        QPointer<CvTrackbar> t = icvFindTrackBarByName(name_bar, window_name);
-        if (t)
-        {
-            int minval = t->slider->minimum();
-            maxval = (maxval>minval)?maxval:minval;
-            t->slider->setMaximum(maxval);
-        }
+        int minval = t->slider->minimum();
+        maxval = (maxval>minval)?maxval:minval;
```

Comment:
This check/truncation is buggy too.

Reply:
@alalek That check and the similar [one](https://github.com/opencv/opencv/blob/master/modules/highgui/src/window_QT.cpp#L697) in `cvSetTrackbarMin` are trying to check that the range is valid. If the `new_max < min` then it just sets the `new_max` to the same as `min`.

I am not sure if this is what you meant.

May I suggest removing that check altogether and the similar [check](https://github.com/opencv/opencv/blob/master/modules/highgui/src/window_QT.cpp#L697) in `cvSetTrackbarMin`? The reason is that Qt5 manages range checking under the hood quite nicely. It seems what Qt5 does under the hood is more intuitive than what those checks are trying to do.

What do you think?

Reply:
I believe OpenCV should do that user asks from OpenCV - without unclear internal logic like min/max values truncation.

This users code should not confuse users:
```
setTrackbarMax(..., -5);
setTrackbarMin(..., -10);
// expected range is [-10..-5] , not [-10..0]
```

In GTK backend I [leave](https://github.com/opencv/opencv/commit/9e239c7a1e86bc76d066c165559df32bacc44dd3#diff-922827782cb034256bf426c7e8d1fe59R1470) this code:
```
trackbar->maxval = maxval;
if (trackbar->maxval >= trackbar->minval)
    gtk_range_set_range(GTK_RANGE(trackbar->widget), trackbar->minval, trackbar->maxval);
```

trackbar->minval / trackbar->maxval is fields from OpenCV trackbar wrapper (need to add them into QT `cvTrackbar`).

If QT handles this internally then we could just call QT directly without extra logic.

BTW,
Perhaps we should have `setTrackbarMinMax` API to avoid inconsistent usage, but it is quite large set of changes for now.

Reply:
Excellent point @alalek.

In the case of the QT4/5, the backend handling appears to do exactly what you desire. So I think it should be fine to remove those checks.

With those checks removed, when I try
```
setTrackbarMax(..., -5);
setTrackbarMin(..., -10);
// expected range is [-10..-5] , not [-10..0]
```

I get the expected range of [-10..-5].